File: lingting-elasticsearch/src/main/java/live/lingting/framework/elasticsearch/datascope/AllowElasticsearchDataPermissionHandler.java
package live.lingting.framework.elasticsearch.datascope;

import java.util.ArrayList;
import java.util.List;

/**
 * @author lingting 2024-07-01 17:29
 */
public class AllowElasticsearchDataPermissionHandler implements ElasticsearchDataPermissionHandler {

	@Override
	public List<ElasticsearchDataScope> dataScopes() {
		return new ArrayList<>();
	}

	@Override
	public List<ElasticsearchDataScope> filterDataScopes(String index) {
		return new ArrayList<>();
	}

	@Override
	public boolean ignorePermissionControl(String index) {
		return true;
	}

}

File: lingting-elasticsearch/src/main/kotlin/live/lingting/framework/elasticsearch/datascope/AllowElasticsearchDataPermissionHandler.kt
package live.lingting.framework.elasticsearch.datascope

/**
 * @author lingting 2024-07-01 17:29
 */
class AllowElasticsearchDataPermissionHandler : ElasticsearchDataPermissionHandler {
    override fun dataScopes(): List<ElasticsearchDataScope> {
        return listOf<ElasticsearchDataScope>()
    }

    override fun filterDataScopes(index: String): List<ElasticsearchDataScope> {
        return listOf<ElasticsearchDataScope>()
    }

    override fun ignorePermissionControl(index: String): Boolean {
        return true
    }
}
